x-pack/legacy/plugins/epm/server/lib/field.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// This should become a copy of https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/mapping/field.go#L39
+export interface Field {
+  name: string;
+  type: string;
+  required: boolean;
+  description: string;
+  fields: Field[];
+}
+
+type Fields = Field[];
+
+/**
+ * ProcessFields takes the given fields read from yaml and expands it.
+ * There are dotted fields in the field.yml like `foo.bar`. These should
+ * be stored as an object inside an object and is the main purpose of this
+ * preprocessing.
+ */
+export function ProcessFields(fields: Fields) {
+  fields.forEach((field, key) => {
+    const fieldName = field.name;
+
+    // If the field name contains a dot, it means we need to create sub objects
+    if (fieldName.includes('.')) {
+      // Split up the name by dots to extract first and other parts
+      const nameParts = fieldName.split('.');
+
+      // Getting first part of the name for the new field
+      const newNameTop = nameParts[0];
+      delete nameParts[0];
+
+      // Put back together the parts again for the new field name
+      let newName = nameParts.slice(1).join('.');
```

Comment:
You can also do this as:

```suggestion
      let newName = nameParts.length === 1
        ? nameParts[0]
        : nameParts.slice(1).join('.'); 
```